Write tests for boolean group broken

binarygit opened the pull request:
# Description
<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->


<!--
  By submitting the Contribution, you acknowledge that you have read the Contributor License Agreement at https://avohq.io/cla and agree to be bound by its terms.
-->
This PR adds tests for the broken boolean group fix PR. 

It also makes sure such boolean group's check boxes are correctly checked in the edit view.

![annotely_image](https://github.com/user-attachments/assets/b4646fa1-148d-4c83-be12-addb3bbae79f)


# Checklist:
<!--
  Please go through the steps and complete them if they make sense (add tests if the change requires it, add to the docs, etc.)
  (Mark [x] inside the brackets)
-->

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Screenshots & recording
<!-- "A picture is worth a thousand words." A video, ten thousand. -->
<!-- Can the behavior touched in this PR be displayed as a screenshot ro a recording. Please attach it. It makes the review easier to pass. -->

## Manual review steps
<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. -->

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

binarygit commented on spec/dummy/app/models/user.rb:
```diff
@@ -97,8 +99,4 @@ def permissions
     }
   end
-
-  def permissions=(value)
-    # assert here the value is a hash and has the right values sent in by the user
-  end
```
@adrianthedev I thought it'd be nicer to not have a body for this at all and just have this method created via the setter. WDYT?

adrianthedev replied:
Oh, so basically set the value on the model instance class. 
Yeah, that could work too. 
Please as a comment on the writer why it's there.

Paul-Bob commented:
@binarygit, Let's keep the old tests for roles to ensure nothing breaks. Let's instead add some new test cases for permissions without modifying the roles tests.

binarygit commented:
All the old tests are there. I just refactored them a little bit and added the new tests but the diff is showing that I changed the whole thing for some reason.

adrianthedev reviewed (approved):
Looks good, thank you!
